src/datasets/load.py

```diff
@@ -1678,8 +1652,14 @@ def load_dataset_builder(
     builder_kwargs = dataset_module.builder_kwargs
     data_dir = builder_kwargs.pop("data_dir", data_dir)
     data_files = builder_kwargs.pop("data_files", data_files)
-    config_name = builder_kwargs.pop("config_name", name)
+    config_name = builder_kwargs.pop("config_name", name or dataset_module.default_config_name or "default")
```

Comment:
this causes failing of loading datasets with a single config **without** providing config name, see [this test](https://github.com/huggingface/datasets/pull/5331/files#diff-39f545cd21f1ca75281b3f357c296be83c964764f99dc24888f9e1e3f8fa05c6R1139). Because it passes "default" while should pass None in this case.

basically, "default" should be passed only for non-configured packaged loaders (without metadata)